sources/editor/Stride.GameStudio/ViewModels/DebuggingViewModel.cs

```diff
@@ -162,50 +162,56 @@ private async void PullAssemblyChanges([NotNull] ProjectWatcher projectWatcher)
-                    if (assemblyChange.ChangeType == AssemblyChangeType.Binary)
-                        continue;
-
-                    var shouldNotify = !assemblyChangesPending;
-                    modifiedAssemblies[assemblyChange.Assembly] = new ModifiedAssembly
+                    await WaitUntil(() => projectWatcher.Events != null);
+                    var events = projectWatcher.Events;
+                    projectWatcher.Events = null;
+                    foreach(var assemblyChange in  events)
                     {
-                        LoadedAssembly = assemblyChange.Assembly,
-                        ChangeType = assemblyChange.ChangeType,
-                        Project = assemblyChange.Project
-                    };
-
-                    Dispatcher.Invoke(() =>
-                    {
-                        UpdateCommands();
-
-                        if (shouldNotify)
+                        if (assemblyChange == null || assemblyChange.ChangeType == AssemblyChangeType.Binary)
+                            continue;
+                        modifiedAssemblies[assemblyChange.Assembly] = new ModifiedAssembly
                         {
-                            var message = Tr._p("Message", "Some game code files have been modified. Do you want to reload the assemblies?");
-                            ServiceProvider.Get<IEditorDialogService>().AddDelayedNotification(EditorSettings.AskBeforeReloadingAssemblies, message,
-                                Tr._p("Button", "Reload"), Tr._p("Button", "Don't reload"),
-                                yesAction: async () =>
-                                {
-                                    var undoRedoService = ServiceProvider.Get<IUndoRedoService>();
-                                    // Wait for current transactions, undo/redo or save to complete before continuing.
-                                    await Task.WhenAll(undoRedoService.TransactionCompletion, undoRedoService.UndoRedoCompletion, Session.SaveCompletion);
-                                    // Reload assembly, if possible
-                                    if (ReloadAssembliesCommand.IsEnabled)
-                                        ReloadAssembliesCommand.Execute();
-                                },
-                                yesNoSettingsKey: EditorSettings.AutoReloadAssemblies);
-                        }
-                    });
+                            LoadedAssembly = assemblyChange.Assembly,
+                            ChangeType = assemblyChange.ChangeType,
+                            Project = assemblyChange.Project
+                        };
+                    }
+
+                    var shouldNotify = !assemblyChangesPending;
+                    if (modifiedAssemblies.Count > 0 && shouldNotify)
+                        RunNotifier(true);
                 }
             }
         }
 
+        private async void RunNotifier(bool shouldNotify)
```

Comment:
Refrain from using `async void`.

Reply:
https://github.com/stride3d/stride/pull/2106/commits/5ad7e3dfd9029977b2bdd7b3ef8a5605561424b3
it was indeed not even needed